### PR TITLE
logs: reduce number of dependencies for the logs client

### PR DESCRIPTION
**Pull request by iksaif**
### What does this PR do?

This will allow us to import it elsewhere.

### Motivation

https://github.com/DataDog/dd-go/pull/8764

client's depdencies
godepgraph 3

sender's dependencies
godepgraph 4

(from `godepgraph -nostdlib github.com/DataDog/datadog-agent/pkg/logs/sender | dot -Tpng -o godepgraph.png`)

### CODE REVIEW

File: pkg/logs/status/status.go [hunk 1]
```diff
@@ -6,6 +6,8 @@
 package status
 
 import (
+	"expvar"
+	"github.com/DataDog/datadog-agent/pkg/logs/metrics"
```
Comment (ajacquemot) [inline pkg/logs/status/status.go]: nits, can you move this at the very end of the list of dependencies ?

File: pkg/logs/status/status_test.go [hunk 2]
```diff
@@ -70,3 +71,8 @@ func TestStatusDeduplicateWarnings(t *testing.T) {
 	status := Get()
 	assert.ElementsMatch(t, []string{"Identical Warning", "Unique Warning"}, status.Messages)
 }
+
+func TestMetrics(t *testing.T) {
```
Comment (ajacquemot) [inline pkg/logs/status/status_test.go]: Can you add more tests please to assert that:
- adding new messages increase the number of warnings
- setting up the agent change the value of `IsRunning`

File: pkg/logs/status/status.go [hunk 3]
```diff
@@ -137,3 +139,12 @@ func toDictionary(c *config.LogsConfig) map[string]interface{} {
 	}
 	return dictionary
 }
+
+func init() {
```
Comment (ajacquemot) [inline pkg/logs/status/status.go]: go-noob, wondering whether those functions always return the value of the init. Let's write some tests to assert that this works properly.

**Review by ajacquemot — CHANGES_REQUESTED** [review]
LGTM, left nits and one change request.

File: pkg/logs/status/status.go (hunk 1 above)
Comment (iksaif) [reply pkg/logs/status/status.go]: done

File: pkg/logs/status/status_test.go (hunk 2 above)
Comment (iksaif) [reply pkg/logs/status/status_test.go]: done

File: pkg/logs/status/status.go (hunk 3 above)
Comment (iksaif) [reply pkg/logs/status/status.go]: i'm not sure what you mean here ? They do not return anything
More docs at https://golang.org/doc/effective_go.html#init

**Review by ajacquemot — APPROVED** [review]

**Comment by arbll** [comment]
Let's not delete this branch for black-magic reasons

**Pull request merged**